lib/book/book_page/bloc/book_page_bloc.dart

```diff
@@ -175,4 +170,23 @@ class BookPageBloc extends Bloc<BookPageEvent, BookPageState> {
           error: e, bookData: book.bookData!, pageIndex: currentPageIndex));
     }
   }
+
+  Future<void> _onDeleteBookPage(DeleteBookPageEvent event, Emitter<BookPageState> emit) async{
+    emit(LoadingState());
+    try{
+      await FirebaseDbManager.instance.removePage(book.bookData!.bookPages[currentPageIndex], book.docId);
```

Comment:
Nothing wrong here but instead of method to remove page, you can simply update all pages on server after updating the page numbers locally,